src/service/viewport-impl.js

```diff
@@ -686,9 +687,9 @@ export class ViewportBindingNatural_ {
   }
 
   /** @override */
-  getLayoutRect(el) {
-    const scrollTop = this.getScrollTop();
-    const scrollLeft = this.getScrollLeft();
+  getLayoutRect(el, opt_scrollLeft, opt_scrollTop) {
```

Comment:
Let's add a comment to `onScroll` to explain that cached value must be updated or invalidated.